### PR TITLE
Convert ScalarMassElemKernel for GPU operation.

### PR DESCRIPTION
**Pull request by alanw0**

### CODE REVIEW

File: include/kernel/ScalarMassElemKernel.h [hunk 1]
```diff
@@ -73,8 +75,7 @@ class ScalarMassElemKernel: public Kernel
   /// Integration point to node mapping
   const int* ipNodeMap_;
```
Comment (sayerhs) [inline include/kernel/ScalarMassElemKernel.h]: Suggest removing the unused pointer

File: unit_tests/kernels/UnitTestKernelUtils.h [hunk 2]
```diff
@@ -536,6 +540,7 @@ class MixtureFractionKernelHex8Mesh : public TestKernelHex8Mesh
   }
 
   ScalarFieldType* mixFraction_{nullptr};
+  ScalarFieldType* scalarQ_{nullptr};
```
Comment (sayerhs) [inline unit_tests/kernels/UnitTestKernelUtils.h]: `mixFraction_` is the `scalarQ_` that we should be using. No need to introduce another field here.

File: unit_tests/kernels/UnitTestScalarMassElem.C [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+  // Populate LHS and RHS
+  helperObjs.assembleElemSolverAlg->execute();
+
+#ifndef KOKKOS_ENABLE_CUDA
+  EXPECT_EQ(helperObjs.linsys->lhs_.extent(0), 8u);
+  EXPECT_EQ(helperObjs.linsys->lhs_.extent(1), 8u);
+  EXPECT_EQ(helperObjs.linsys->rhs_.extent(0), 8u);
+
+//  std::cerr.precision(14);
```
Comment (sayerhs) [inline unit_tests/kernels/UnitTestScalarMassElem.C]: Do we need to keep around this commented section? Might be useful to add a helper method in `HelperObjects` that could print out the lhs/rhs on demand (could be used elsewhere).

File: include/kernel/ScalarMassElemKernel.h (hunk 1 above)
Comment (alanw0) [reply include/kernel/ScalarMassElemKernel.h]: Dang, how did I miss that?

File: unit_tests/kernels/UnitTestKernelUtils.h (hunk 2 above)
Comment (alanw0) [reply unit_tests/kernels/UnitTestKernelUtils.h]: I needed to make it 2 states, so I made another field to make sure I wouldn't break anything else. But I can certainly make mixFraction be 2 states instead.

File: unit_tests/kernels/UnitTestScalarMassElem.C (hunk 3 above)
Comment (alanw0) [reply unit_tests/kernels/UnitTestScalarMassElem.C]: Yes I'll clean that up. I meant to do that already, and forgot.

**Comment by alanw0** [comment]
@sayerhs I've now pushed a commit to address your comments.

**Pull request merged**